packages/renderer/src/lib/image/BuildImageFromContainerfile.svelte

```diff
@@ -122,10 +122,20 @@ async function getContainerfileLocation() {
   const result = await window.openFileDialog('Select Containerfile to build');
   if (!result.canceled && result.filePaths.length === 1) {
     containerFilePath = result.filePaths[0];
+    hasInvalidFields = false;
+
+    // normalize the file path to posix style
+    // eslint-disable-next-line no-useless-escape
+    const posixContainerDirPath = containerFilePath.replace(/\\/g, '/').replace(/\/[^\/]*$/, '');
+    const containerDirName = posixContainerDirPath.split('/').slice(-1)[0];
+    if (containerDirName) {
```

Comment:
should we set the image name only if it's not already defined ?

I see pro/cons here

Reply:
+1 from me, but I would change it slightly to 'was edited by the user'. i.e. if the user hasn't edited the field and it was auto-generated, then they pick a different path, we should auto-generate it again. But if the user typed in the name, or edited one we auto-generated, we shouldn't change it again. It is probably safe enough to say 'focussed on the field' = 'user edited' for coding simplicity.

I've seen this in other products and it strikes a good balance between being very helpful but not overwriting anything the user has done.

Reply:
hey guys, thanks for the input. Will try to implement it like you described when I find the time 👍